### PR TITLE
Update x-on, $event and $dispatch docs.

### PR DESCRIPTION
**Pull request by atomgiant**
Added an example of how to access $event and $dispatch inside of a
Javascript function.  This also includes adding an example of passing in
a Javascript name to x-on.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -562,6 +574,12 @@ You can also use `$dispatch()` to trigger data updates for `x-model` bindings. F
 </div>
 ```
 
+> Note: The $dispatch property is only available in the DOM.
+
+If you need to access $dispatch inside of a Javascript function you can pass it in directly:
```
Comment (HugoDF) [inline README.md:579]:
```suggestion
If you need to access $dispatch inside of a JavaScript function you can pass it in directly:
```

File: README.md [hunk 2]
```diff
@@ -536,6 +542,12 @@ These behave exactly like VueJs's transition directives, except they have differ
 
 `$event` is a magic property that can be used within an event listener to retrieve the native browser "Event" object.
 
+> Note: The $event property is only available in the DOM.
+
+If you need to access $event inside of a Javascript function you can pass it in directly:
```
Comment (HugoDF) [inline README.md:547]:
```suggestion
If you need to access $event inside of a JavaScript function you can pass it in directly:
```

File: README.md (hunk 1 above)
Comment (atomgiant) [reply README.md]: Thanks @HugoDF 👍

**Comment by calebporzio** [comment]
Thanks!

**Pull request merged**